molecule/read-certs-secrets-test/converge.yml

```
@@ -0,0 +1,46 @@
+- name: Tests
+  hosts: localhost
+  connection: local
+  collections:
+  - kubernetes.core
+  vars:
+    custom_resource: "{{ lookup('template', cr_file_path) | from_yaml }}"
+  tasks:
+  - import_tasks: ../common/tasks.yml
+  - import_tasks: ../asserts/pod_asserts.yml
+  - import_tasks: ../common/wait_for_kiali_running.yml
+  
+  - name: Get control plane role
+    vars:
+      instance_name: "{{ kiali.instance_name | default('kiali') }}"
+    k8s_info:
+      api_version: rbac.authorization.k8s.io/v1
+      kind: Role
+      name: "{{ instance_name }}-controlplane"
+      namespace: "{{ istio.control_plane_namespace }}"
+    register: controlplane_role
+
+  - name: Gather the rules that contains resource names
+    set_fact:
+      rules: "{{ controlplane_role.resources[0].rules | selectattr('resourceNames','defined') }}"
+
+  - name: Get the needed role for reading certificates
+    vars:
+      istio_secret: "istio-ca-secret"
+      user_secret: "cacerts"
+    set_fact:
+      cert_role: "{{ item }}"
+    with_items: "{{ rules }}"
+    when: "{{ istio_secret in item.resourceNames and user_secret in item.resourceNames }}"
+
+  - name: Check that the role has the right verbs
+    vars:
+      get_verb: "get"
+      list_verb: "list"
+      watch_verb: "watch"
+    assert:
+      that: 
+      - cert_role is defined
+      - get_verb in cert_role.verbs
+      - list_verb in cert_role.verbs
+      - watch_verb in cert_role.verbs
```

I would also confirm that those are the ONLY 3 verbs (ensures no one accidentally gives "patch" or "delete", for example):
```suggestion
      - watch_verb in cert_role.verbs
      - cert_role.verbs | length == 3
```